src/wp-includes/media.php

```diff
@@ -5449,3 +5451,285 @@ function wp_increase_content_media_count( $amount = 1 ) {
+		$data        .= sprintf( 'data-has-transparency="%s" ', $transparency );
+		$extra_class  = $image_meta['has_transparency'] ? 'has-transparency' : 'not-transparent';
+	}
+
+	if ( ! empty( $data ) || ! empty( $style ) || ! str_contains( $style, $filtered_image ) || ! str_contains( $data, $filtered_image ) ) {
+		$filtered_image = str_replace( '<img ', '<img ' . $data . $style, $filtered_image );
+	}
+	if ( ! empty( $extra_class ) ) {
+		$filtered_image = str_replace( ' class="', ' class="' . $extra_class . ' ', $filtered_image );
+	}
+
+	return $filtered_image;
+}
+
+/**
+ * Add CSS needed for to show the dominant color as an image background.
+ *
+ * @since 6.1
+ */
+function dominant_color_add_inline_style() {
+	$handle = 'dominant-color-styles';
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = 'img[data-dominant-color]:not(.has-transparency) { background-color: var(--dominant-color); background-clip: content-box, padding-box; }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+add_filter( 'wp_enqueue_scripts', 'dominant_color_add_inline_style' );
+
+/**
+ * Computes the Dominant color of the given attachment image.
+ *
+ * @since 6.1
+ * @access private
+ *
+ * @param int $attachment_id The attachment ID.
+ * @return array|WP_Error Array with the Dominant color and has transparency values or WP_Error on error.
+ */
+function _dominant_color_get_dominant_color_data( $attachment_id ) {
+	$mime_type = get_post_mime_type( $attachment_id );
+	if ( 'application/pdf' === $mime_type ) {
+		return new WP_Error( 'no_image_found', __( 'Unable to load image.' ) );
+	}
+	$file = wp_get_attachment_file_path( $attachment_id );
+	if ( ! $file ) {
+		$file = get_attached_file( $attachment_id );
+	}
+
+	$editor = wp_get_image_editor(
+		$file,
+		array(
+			'methods' => array(
+				'get_dominant_color',
+				'has_transparency',
+			),
+		)
+	);
+
+	if ( is_wp_error( $editor ) ) {
+		return $editor;
+	}
+
+	// can the image type be transparent if not lets shortcut here
+	if ( in_array( $mime_type, array( 'image/jpeg', 'image/bmp' ), true ) ) {
+		$dominant_color_data['has_transparency'] = false;
+	} else {
+		$has_transparency = $editor->has_transparency();
+		if ( is_wp_error( $has_transparency ) ) {
+			return $has_transparency;
+		}
+		$dominant_color_data['has_transparency'] = $has_transparency;
+	}
+
+	$dominant_color = $editor->get_dominant_color();
+	if ( is_wp_error( $dominant_color ) ) {
+		return $dominant_color;
+	}
+	$dominant_color_data['dominant_color'] = $dominant_color;
+
+	return $dominant_color_data;
+}
+
+/**
+ * Gets file path of image based on size.
+ *
+ * @since 6.1
+ *
+ * @param int    $attachment_id Attachment ID for image.
+ * @param string $size          Optional. Image size. Default 'thumbnail'.
+ * @return false|string Path to an image or false if not found.
+ */
+function wp_get_attachment_file_path( $attachment_id, $size = 'medium' ) {
```

Comment:
Mismatch between default value for `$size` and the PHPDoc (`medium` vs `thumbnail`)

Reply:
Good spot and thanks
Fixed